### PR TITLE
EIP1-3229: EIP1:4572: Updates GET API spec for Getting AEDs with applicationId as queryParam

### PR DESCRIPTION
**Pull request by vishalgupta-valtech**
image

### CODE REVIEW

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 1]
```diff
@@ -583,6 +583,7 @@ paths:
         uri: '${base_uri}/eros/{eroId}/anonymous-elector-documents'
         requestParameters:
           integration.request.path.eroId: method.request.path.eroId
+          integration.request.querystring.applicationId: method.request.querystring.applicationId
```
Comment (vishalgupta-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:586]: Reference : `https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-method-settings-execution-console.html` for `query` string `applicationId`. 

We might need to do the same for other GET endpoints

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 2]
```diff
@@ -564,11 +564,11 @@ paths:
...
         - Anonymous Elector Documents (AED)
       parameters:
-        - name: eroId
-          description: The ID of the Electoral Registration Office responsible for the application.
+        - name: applicationId
+          description: An identifier of an Anonymous Voter Card Application as a `query param`
           schema:
             type: string
-          in: path
+          in: query
```
Comment (vishalgupta-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:571]: ℹ️ It's a query param instead of path.

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 3]
```diff
@@ -756,28 +758,18 @@ components:
         - printRequestSummaries
     PrintRequestSummary:
       title: PrintRequestSummary
-      type: object
       description: An object containing print request summary
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/UpdatedBy'
```
Comment (vishalgupta-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:764]: I have introduced a new mandatory object type that can now be reused across both summary responses
ℹ️ The response structure hasn't changed, so this is not a breaking change

**Review by sentioservices — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/dluhc/printapi/rest/AnonymousElectorDocumentController.kt [hunk 4]
```diff
@@ -52,6 +57,16 @@ class AnonymousElectorDocumentController(
         }
     }
 
+    @GetMapping
+    @PreAuthorize(HAS_ERO_VC_ANONYMOUS_ADMIN_AUTHORITY)
+    @ResponseStatus(HttpStatus.OK)
```
Comment (Neil-Massey) [inline src/main/kotlin/uk/gov/dluhc/printapi/rest/AnonymousElectorDocumentController.kt:62]: Maybe import

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 5]
```diff
@@ -352,7 +353,7 @@ paths:
       description: Returns the Temporary Certificate summaries for a Voter Card Application
       parameters:
         - name: applicationId
-          description: The identifier of the Voter Card Application.
+          description: The identifier of the Voter Card Application as a `query param`.
```
Comment (Neil-Massey) [inline src/main/resources/openapi/PrintAPIs.yaml:356]: I don't think " as a `query param`" is needed in the description - the schema makes that very clear with the `in: query`

**Review by matt-wills-valtech — APPROVED** [review]
LGTM

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 6]
```diff
@@ -370,7 +371,7 @@ paths:
         uri: '${base_uri}/eros/{eroId}/temporary-certificates'
         requestParameters:
           integration.request.path.eroId: method.request.path.eroId
-          integration.request.path.applicationId: method.request.path.applicationId
+          integration.request.querystring.applicationId: method.request.querystring.applicationId
```
Comment (Neil-Massey) [inline src/main/resources/openapi/PrintAPIs.yaml:374]: Is this needed?  (When Nathan added other query parameters I checked VCA spec and it didn't look like we specifically mapped query parameters)

Comment (vishalgupta-valtech) [reply src/main/resources/openapi/PrintAPIs.yaml:374]: I am unsure 🤷🏼‍♂️  as applicationId is definitely not a `path` variable, so just corrected it to be a `querystring` param.

File: src/main/resources/openapi/PrintAPIs.yaml (hunk 5 above)
Comment (matt-wills-valtech) [reply src/main/resources/openapi/PrintAPIs.yaml:356]: I don't think it does any harm, especially as `applicationId` is in the path of the vast majority of our endpoints.

File: src/main/resources/openapi/PrintAPIs.yaml (hunk 6 above)
Comment (matt-wills-valtech) [reply src/main/resources/openapi/PrintAPIs.yaml:374]: I'm not sure either, but I'd prefer to have consistency with the other relevant endpoints as well.

File: src/main/resources/openapi/PrintAPIs.yaml (hunk 1 above)
Comment (Neil-Massey) [reply src/main/resources/openapi/PrintAPIs.yaml:587]: I think if the query parameter name is the same in the AWS API GW and the endpoint the request is forwarded the passthrough behaviour means no explicit mapping is required.

File: src/main/resources/openapi/PrintAPIs.yaml (hunk 6 above)
Comment (NathanRussellValtech) [reply src/main/resources/openapi/PrintAPIs.yaml:374]: 100% that it needs removing as a path variable (I must have forgotten that when I did it 😬 )
But I'm not sure about adding it as a querystring - I'm not sure it's needed ?
(When I made this change here in print-api to make this a query string param, I took my steer from the Search Application's endpoint in VCA ..... that endpoint has loads of query string params, and we dont define them in the APIG config)

Comment (vishalgupta-valtech) [reply src/main/resources/openapi/PrintAPIs.yaml]: Thank you all . I have removed these and `applicationId queryString` won't be mapped in APIG integration as we all agreed/discussed. PR updated 👍🏼

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 7]
```diff
@@ -370,7 +370,6 @@ paths:
         uri: '${base_uri}/eros/{eroId}/temporary-certificates'
         requestParameters:
           integration.request.path.eroId: method.request.path.eroId
-          integration.request.path.applicationId: method.request.path.applicationId
```
Comment (NathanRussellValtech) [inline src/main/resources/openapi/PrintAPIs.yaml:373]: 👍 Thank you

**Review by NathanRussellValtech — APPROVED** [review]
LGTM 👍

**Pull request merged**